pyproject.toml

```diff
@@ -0,0 +1,38 @@
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "fme"
+version = "2024.10.3"
+description = "Train and evaluate weather/climate model emulators"
+readme = "README.md"
+requires-python = ">=3.9"
+license = {file = "LICENSE"}
```

Comment:
I had to move the `pyproject.toml` to the top level in order to dynamically link the `README` and `LICENSE` files.  I did try to leave it under `fme`, but it's not permitted to access parent directories for dynamic builds.

@mcgibbon doesn't recall if there was a requirement that `setup.py` and other files exist in `fme` and not at the top level, but did mention that it would make things more difficult for having multiple packages.  Could be something worth discussing.

Reply:
This does fundamentally change our repo so that it is now the repo of a single python package, as opposed to being a repo containing a single python package (but which could contain more). Probably worth discussing in our weekly fme sync.

Reply:
I would lean to maintaining the current repo structure. Maybe we could duplicate the LICENSE within `fme` and make an `fme`-specific README within that folder? The benefit of current structure is we can have the scripts folder with data processing etc scripts that are distinct from the fme package.